src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/DisposeAnalysis/DisposeAnalysis.DisposeDataFlowOperationVisitor.cs

```diff
@@ -104,6 +104,13 @@ protected override DisposeAbstractValue HandleInstanceCreation(IOperation creati
                     return defaultValue;
                 }
 
+                // Special case: MemoryStream doesn't need to be disposed. Subclasses *might* need to be disposed, but that is the less common case, and the common case is a huge source of noisy warnings.
+                if (MemoryStreamNamedType != null &&
+                    instanceType.Equals(MemoryStreamNamedType))
```

Comment:
Null check is unnecessary here. I believe `Equals` method checks and handles it correctly.

Reply:
@Youssef1313 I wondered about this, but was initially following the pattern of other null checks on named types in the code. Your feedback made me curious so I wondered why...

The main reason seems to be that the contract of GetOrCreateTypeByMetadataName does not guarantee it will not retrurn null.

So I propose not to fix this feedback..

Reply:
`GetOrCreateTypeByMetadataName` can indeed return null. The thing is that `instanceType.Equals(null)` will then return false. But since you're following existing code pattern, it's more appropriate for this to be a separate PR (unless the early null check is known to be faster than `instanceType.Equals(null)` and has a noticeable performance impact - but I doubt this is true).

Reply:
@Youssef1313 Sounds like the germs of some new code analysis rules in the making... 
"Do not bother checking for null before passing parameters to 'Equals', it usually will just return false anyway."
"Check parameters for null in Equals"